File: src/main/java/com/github/unchama/seichiassist/commands/LastQuitCommand.kt
package com.github.unchama.seichiassist.commands

import arrow.core.some
import com.github.unchama.contextualexecutor.asNonBlockingTabExecutor
import com.github.unchama.contextualexecutor.builder.ContextualExecutorBuilder
import com.github.unchama.contextualexecutor.builder.Parsers
import com.github.unchama.seichiassist.SeichiAssist

object LastQuitCommand {
  val executor = ContextualExecutorBuilder.beginConfiguration()
      .argumentsParsers(listOf(Parsers.identity))
      .execution { context ->
        val playerName = context.args.parsed[0] as String

        SeichiAssist.databaseGateway.playerDataManipulator.inquireLastQuitOf(playerName).some()
      }
      .build()
      .asNonBlockingTabExecutor()
}

File: src/main/scala/com/github/unchama/seichiassist/commands/LastQuitCommand.scala
package com.github.unchama.seichiassist.commands

import com.github.unchama.contextualexecutor.builder.{ContextualExecutorBuilder, Parsers}
import com.github.unchama.seichiassist.SeichiAssist
import org.bukkit.command.TabExecutor

object LastQuitCommand {
  val executor: TabExecutor = ContextualExecutorBuilder.beginConfiguration()
    .argumentsParsers(List(Parsers.identity))
    .execution { context =>
      val playerName = context.args.parsed(0).asInstanceOf[String]

      SeichiAssist.databaseGateway.playerDataManipulator.inquireLastQuitOf(playerName)
    }
    .build()
    .asNonBlockingTabExecutor()
}